File: packages/contracts/typescript/src/telemetry.ts
/**
 * Telemetry correlation metadata for end-to-end observability (spec §9).
 */

/** Telemetry correlation metadata propagated across CLI, API, and services. */
export interface TelemetryContext {
  /** OpenTelemetry trace ID. */
  trace_id: string;
  /** OpenTelemetry span ID. */
  span_id: string;
  /** Platform correlation ID (UUID). */
  correlation_id: string;
  /** Parent span ID for nested operations. */
  parent_span_id?: string | null;
}

File: packages/contracts/src/m2la_contracts/telemetry.py
"""Telemetry correlation metadata for end-to-end observability (spec §9)."""

from pydantic import BaseModel, Field


class TelemetryContext(BaseModel):
    """Telemetry correlation metadata propagated across CLI, API, and services."""

    trace_id: str = Field(..., description="OpenTelemetry trace ID")
    span_id: str = Field(..., description="OpenTelemetry span ID")
    correlation_id: str = Field(..., description="Platform correlation ID (UUID)")
    parent_span_id: str | None = Field(default=None, description="Parent span ID for nested operations")
